Application/components/FriendsColumn/RemoveFriendModal.tsx

```diff
@@ -0,0 +1,148 @@
+import { useUser } from '@clerk/nextjs';
+import { useCache } from '@/contexts/queryCacheContext';
+import { useFriendList } from '@/hooks/useFriendList';
+import { showNotification } from '@mantine/notifications';
+
+interface Friend {
+  id: string;
+  username: string;
+}
+
+interface RemoveFriendResponse {
+  message: string;
+}
+
+export function RemoveFriendModal(): JSX.Element {
+  const { user } = useUser();
+  const theme = useMantineTheme();
+  const [opened, setOpened] = useState<boolean>(false);
+  const { lastFetched, setLastFetched } = useCache();
+  const [selectedFriend, setSelectedFriend] = useState<string[]>([]);
+  const [errorMessages, setErrorMessages] = useState<{ friends: string }>({ friends: '' });
+  const { list: friendsList } = useFriendList({ lastFetched, setLastFetched });
+
+  const friendOptions = friendsList.map((friend: Friend) => ({
+    value: friend.id,
+    label: friend.username,
+  }));
+
+  const handleOpenModalClick = () => {
+    if (friendsList.length > 0) {
+      setOpened(true);
+    } else {
+      showNotification({
+        title: 'No Friends to Remove',
+        message: 'You currently have no friends to remove.😢',
+        color: 'red',
+      });
+    }
+  };
+
+  const handleRemoveFriendClick = async (): Promise<void> => {
+    if (selectedFriend.length === 0) {
+      setErrorMessages({ friends: "Please select a friend to remove." });
+      return;
+    }
+
+    try {
+      const response = await fetch('/api/Remove-friend', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          userID: user?.id,
+          friendIDs: selectedFriend,
+        }),
+      });
+
+      if (response.ok) {
+        // Iterate over selectedFriend array and show a notification for each
+        selectedFriend.forEach(friendId => {
+          const friendName = friendOptions.find(option => option.value === friendId)?.label || 'Unknown friend';
+          showNotification({
+            title: 'Friend Removed',
+            message: `✅ ${friendName} was removed successfully.`,
+            color: 'green',
+          });
+        });
+
+        setSelectedFriend([]);
+        setErrorMessages({ friends: '' });
+        setLastFetched(Date.now());
+        setOpened(false);
+      } else {
+        const errorText: string = await response.text();
+        showNotification({
+          title: 'Error',
+          message: errorText || 'Failed to remove friend.',
+          color: 'red',
+        });
+      }
+    } catch (error: any) {
+      console.error('Error removing friend:', error);
+      showNotification({
+        title: 'Error',
+        message: error.toString() || 'An error occurred while removing the friend.',
+        color: 'red',
+      });
+    }
+  };
+
+  return (
+    <>
+      <Button
+        onClick={handleOpenModalClick}
+        variant="subtle"
+        style={{
```

Comment:
This entire inline CSS styling needs to go. The linear-gradient can be specified just by passing in the field `variant: "gradient"` as the gradient is globally configured to be the nice pink-orange transition within the RootLayout. That's how you access it. Also, never hardcode pixels. Use "sm" or "md" for example for the padding and radius, as those are provided by `Mantine` and ensure uniform styling across all components.